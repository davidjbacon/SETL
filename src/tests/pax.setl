program m;
use p1, p2;
var y;
init z2 := 'maps';
print(f1(),f2());
print(x1,x2);
print(z1,p1.z1,z2,p2.z2,m.z2);
y:=5;
print(y,m.y,p1.y,p2.y);
end m;

package p1;
const x1 = {to_upper x : x in 'spam 1'};
var y := 1;
init z1 := 'spam';
proc f1;
end p1;
package body p1;
const a1 = {reverse s : s in x1};
var b1;
init c1 := 'more spam';
var d1 := f1();
proc f1; return g1(); end;
proc g1; print(d1); return a1; end;
end p1;

package p2;
const x2 = {to_lower x : x in 'spam 2'};
var y := 2;
init z2 := 'spam';
proc f2;
end p2;
package body p2;
use p1;
const a2 = {pretty s : s in x2} + {pretty s : s in x1};
var b2;
init c2 := 'more spam';
var d2 := g2();
proc f2; return g2(); end;
proc g2; print(d2); return a2; end;
end p2;
