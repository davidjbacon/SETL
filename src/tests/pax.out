*
*
{' ' '1' A M P S}
{''' ''' '''1''' '''2''' '''A''' '''M''' '''P''' '''S''' '''a''' '''m''' '''p''' '''s'''}
{' ' '1' A M P S} {''' ''' '''1''' '''2''' '''A''' '''M''' '''P''' '''S''' '''a''' '''m''' '''p''' '''s'''}
{' ' '1' A M P S} {' ' '2' a m p s}
spam spam maps spam maps
5 5 1 2
